src/modules/rewards/services/opRebateServiceV2.ts

```diff
@@ -0,0 +1,347 @@
+    @InjectRepository(OpRewardV2) readonly rewardRepository: Repository<OpRewardV2>,
+    private marketPriceService: MarketPriceService,
+    private ethProvidersService: EthProvidersService,
+    private appConfig: AppConfig,
+    private dataSource: DataSource,
+  ) {}
+
+  public async getEarnedRewards(userAddress: string) {
+    userAddress = assertValidAddress(userAddress);
+
+    const baseQuery = this.buildBaseQuery(this.rewardRepository.createQueryBuilder("r"), userAddress);
+    const { opRewards } = await baseQuery
+      .select("SUM(CAST(r.amount as DECIMAL))", "opRewards")
+      .where("r.isClaimed = :isClaimed", { isClaimed: true })
+      .getRawOne<{ opRewards: string }>();
+
+    return opRewards;
+  }
+
+  public async getOpRebatesSummary(userAddress: string) {
+    userAddress = assertValidAddress(userAddress);
+
+    const baseQuery = this.buildBaseQuery(this.rewardRepository.createQueryBuilder("r"), userAddress);
+    baseQuery.andWhere("r.isClaimed = :isClaimed", { isClaimed: false });
+    const [{ depositsCount }, { unclaimedRewards }, { volumeUsd }] = await Promise.all([
+      baseQuery.select("COUNT(DISTINCT r.depositPrimaryKey)", "depositsCount").getRawOne<{
+        depositsCount: string;
+      }>(),
+      baseQuery.select("SUM(CAST(r.amount as DECIMAL))", "unclaimedRewards").getRawOne<{
+        unclaimedRewards: number;
+      }>(),
+      baseQuery
+        .leftJoinAndSelect("r.deposit", "d")
+        .leftJoinAndSelect("d.token", "t")
+        .leftJoinAndSelect("d.price", "p")
+        .select(`COALESCE(SUM(d.amount / power(10, t.decimals) * p.usd), 0)`, "volumeUsd")
+        .getRawOne<{
+          volumeUsd: number;
+        }>(),
+      // TODO: add claimable rewards
+    ]);
+
+    return {
+      depositsCount: parseInt(depositsCount),
+      unclaimedRewards,
+      volumeUsd,
+      claimableRewards: "0",
+    };
+  }
+
+  // public async getOpRebateRewards(query: GetRewardsQuery) {
+  //   const limit = parseInt(query.limit ?? "10");
+  //   const offset = parseInt(query.offset ?? "0");
+  //   const userAddress = assertValidAddress(query.userAddress);
+
+  //   const baseQuery = this.buildBaseQuery(this.rewardRepository.createQueryBuilder("r"), userAddress);
+
+  //   const rewardsQuery = baseQuery.orderBy("r.depositDate", "DESC").limit(limit).offset(offset);
+  //   const [rewards, total] = await rewardsQuery.getManyAndCount();
+     
+  //   // JOIN instead of query deposits separately 
+  //   const depositPrimaryKeys = rewards.map((reward) => reward.depositPrimaryKey);
+  //   const deposits = await this.depositRepository.find({
+  //     where: { id: In(depositPrimaryKeys) },
+  //   });
+
+  //   return {
+  //     rewards: rewards.map((reward) => ({
+  //       ...reward,
+  //       deposit: deposits.find((deposit) => deposit.id === reward.depositPrimaryKey),
+  //     })),
+  //     pagination: {
+  //       limit,
+  //       offset,
+  //       total,
+  //     },
+  //   };
+  // }
```

Comment:
This could be rewritten to use JOIN instead of separate queries now that the FK between rewards and deposits is a composite key (depositId, originChainId).